src/DataDepsGenerators.jl

```diff
@@ -115,11 +116,15 @@ function format_papers(authors::Vector, year::String, name::String, link::String
 end
 
 function match_doi(uri::String)
-    identifier = match(r"\b(10[.][0-9]{4,}(?:[.][0-9]+)*\/(?:(?![\"&\'<>])\S)+)\b", uri).match
-    return identifier
+    try
+        identifier = match(r"\b(10[.][0-9]{4,}(?:[.][0-9]+)*\/(?:(?![\"&\'<>])\S)+)\b", uri).match
+        return true, identifier
+    catch ErrorException
+        return false
+    end
```

Comment:
This is not a nice return value.
`Union{Void, T}` is a standard pattern for return values in julia. (and is becoming more so in 0.7).

`Union{Bool, Tuple{Bool, T}}` is not.

What is throwing an `ErrorException` here? Most things in `Base` should not throw that kind of exception, so it might be worth raising up stream

Reply:
The `ErrorException` is to handle `Void` which doesn't have a match attribute. What would you suggest to do otherwise, in case of returning values. I can do `try catch` instead of `if else` but I don't think its preferred in any way.

Reply:
Huh, I didn't think that threw an `ErrorException` I thought it threw something else.
Anyway for some alternatives see
 https://github.com/oxinabox/DataDepsGenerators.jl/pull/34/files#r197491537

Reply:
Cool, better.
A slightly cleaner way of writing this (so as not to be using exception handling)
is 
```
function match_doi(uri::String)
    identifier_match = match(r"\b(10[.][0-9]{4,}(?:[.][0-9]+)*\/(?:(?![\"&\'<>])\S)+)\b", uri)
    identifier_match === nothing ? nothing : identifier_match.match
end
```

Also I just realized `catch ErrorException`
doesn't do what you think it does.
Julia doesn't have type-guarded catch.
So `catch ErrorException` actually creates a local variable called `ErrorException` that is set to the exception 
value.

What you want to write for guarded exception handling is 

```
catch err
    if err isa ErrorException
         # handle
    else
         rethrow()
     end
end
```